core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -304,6 +304,14 @@ copy_init:
 	bgt	copy_init
 #endif
 
+	/* Clear .bss */
+	ldr	r0, =__bss_start
+	ldr	r1, =__bss_end
+	mov	r2, #0
+clear_bss:
+	strb	r2, [r0], #1
```

Comment:
minor: could speed up a bit through `str r2, [r0], #4`. **kern.ld.S** already include the required alignment for the `__bss_start` and `__bss_end`. From few builds, it seems early bss was below 1kByte, while bss is below 100kB.
Same could apply to **generic_entry_a64.S**.